src/promptflow/promptflow/_sdk/_service/utils/utils.py

```diff
@@ -36,6 +55,35 @@ def get_random_port():
         return s.getsockname()[1]
 
 
+def _get_process_by_port(port):
+    for proc in psutil.process_iter(["pid", "connections", "create_time"]):
+        try:
+            connections = proc.info["connections"]
+        except psutil.AccessDenied:
+            pass
+        else:
+            for connection in connections:
+                if connection.laddr.port == port:
+                    return proc
+
+
+def kill_exist_service(port):
+    proc = _get_process_by_port(port)
+    if proc:
+        proc.kill()
```

Comment:
what's the signal send to process here?

Reply:
it send SIGKILL to process in POSIX. In Windows, it will call function in C extension

Reply:
Maybe we need to do additiaonl sigterm if SIGKILL not gracefully shutdown this guy

Reply:
change to terminate and wait for the process to be terminated.